### PR TITLE
Use namedtuples instead of dicts for app_info

### PR DESCRIPTION
**Pull request by theychx**
:tv:

### CODE REVIEW

**Comment by theychx** [comment]
I was bored, so I made a preemptive strike on your review. :smile:

**Comment by skorokithakis** [comment]
Haha, strike successful! By the way, I wouldn't be opposed to pulling in [attrs](https://github.com/python-attrs/attrs) as a dependency at any time if you think it would be useful.

File: catt/controllers.py [hunk 1]
```diff
@@ -528,9 +536,7 @@ class DefaultCastController(CastController, MediaControllerMixin, PlaybackBaseMi
     def __init__(self, cast, name, app_id, prep=None):
```
Comment (skorokithakis) [inline catt/controllers.py]: These arguments look suspiciously like we're passing an `app: App` in here, should we just pass that object?

Comment (theychx) [reply catt/controllers.py]: Nice catch. Let me fix that.

File: catt/controllers.py [hunk 2]
```diff
@@ -94,13 +94,13 @@ def get_stream(url, device_info=None, host=None, ytdl_options=None):
...
 
-def get_app_info(id_or_name, cast_type=None, strict=False, show_warning=False):
+def get_app(id_or_name, cast_type=None, strict=False, show_warning=False):
     if id_or_name == "default":
         return DEFAULT_APP
 
     arg_is_id = True if re.match("[0-9A-F]{8}$", id_or_name) else False
     try:
-        app_info = next(a for a in APP_INFO_LIST if (a.id if arg_is_id else a.name) == id_or_name)
+        app = next(a for a in APPS if (a.id if arg_is_id else a.name) == id_or_name)
```
Comment (skorokithakis) [inline catt/controllers.py]: The generator/`next` thing here might be a bit too confusing to save space on a list of two elements, but that's what we had before, so it's fine.

File: catt/controllers.py [hunk 3]
```diff
@@ -337,14 +336,14 @@ def wait_for_states(self, timeout=None):
 
 
 class CastController:
-    def __init__(self, cast, name, app_id, prep=None):
+    def __init__(self, cast, app, prep=None):
```
Comment (skorokithakis) [inline catt/controllers.py]: Can we add types here?

File: catt/controllers.py (hunk 2 above)
Comment (theychx) [reply catt/controllers.py]: I actually had an if clause with a generator for each argument type initially. Would you prefer that?

Comment (skorokithakis) [reply catt/controllers.py]: We're just trying to find the app with the given id, right? Why not just a for/else loop with a return?

File: catt/controllers.py (hunk 3 above)
Comment (theychx) [reply catt/controllers.py]: Let's see if Miss Mypy has one of her good days.

Comment (skorokithakis) [reply catt/controllers.py]: :crossed_fingers:

File: catt/controllers.py (hunk 2 above)
Comment (theychx) [reply catt/controllers.py]: Or with the given name. I brewed those together to make calls to the function cleaner.

Comment (skorokithakis) [reply catt/controllers.py]: Ah okay, I'm not fussy about it. I would have done something like:

```
for app in APPS:
    if id_or_name in (app.id, app.name):
        return app
else:
    return DEFAULT_APP  # todo strict handling etc
```

Which I think looks a bit simpler.

File: catt/controllers.py [hunk 4]
```diff
@@ -85,40 +94,38 @@ def get_stream(url, device_info=None, host=None, ytdl_options=None):
     return StreamInfo(url, host=host, model=cc_info, device_type=cast_type, ytdl_options=ytdl_options)
 
 
-def get_app_info(id_or_name, cast_type=None, strict=False, show_warning=False):
+def get_app(id_or_name, cast_type=None, strict=False, show_warning=False):
     if id_or_name == "default":
```
Comment (skorokithakis) [inline catt/controllers.py]: Do we need this? If `id_or_name` is `"default"` the code below is going to return the default app anyway, no?

File: catt/controllers.py (hunk 2 above)
Comment (theychx) [reply catt/controllers.py]: I don't like "for loop else". I think it looks weird.

File: catt/controllers.py (hunk 4 above)
Comment (theychx) [reply catt/controllers.py]: True. I think I just put it there to not engage in any logic if the requester specifically needs "default".

File: catt/controllers.py (hunk 3 above)
Comment (theychx) [reply catt/controllers.py]: Mypy doesn't like me running ```register_handler``` on an attribute that might not be there, so I made that error go away.

**Comment by theychx** [comment]
Any further comments before I merge?

File: catt/controllers.py (hunk 2 above)
Comment (skorokithakis) [reply catt/controllers.py]: You can just not use the else. It'll return if it matches in the loop.

File: catt/controllers.py (hunk 3 above)
Comment (skorokithakis) [reply catt/controllers.py]: Sounds good!

File: catt/controllers.py (hunk 4 above)
Comment (skorokithakis) [reply catt/controllers.py]: Sounds good to me.

**Comment by skorokithakis** [comment]
Nope, looks good to me!

**Pull request merged**